plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -238,47 +285,29 @@ private InternalTableMetadata makeInternalTableMetadata(String schema, String ta
             result.put(field.getName(), new ElasticsearchColumnHandle(
                     field.getName(),
                     toTrinoType(field),
+                    generateHandleProperties(field),
                     supportsPredicates(field.getType())));
         }
 
         return result.build();
     }
 
-    private static boolean supportsPredicates(IndexMetadata.Type type)
-    {
-        if (type instanceof DateTimeType) {
-            return true;
-        }
-
-        if (type instanceof PrimitiveType) {
-            switch (((PrimitiveType) type).getName().toLowerCase(ENGLISH)) {
-                case "boolean":
-                case "byte":
-                case "short":
-                case "integer":
-                case "long":
-                case "double":
-                case "float":
-                case "keyword":
-                    return true;
-            }
-        }
-
-        return false;
-    }
-
     private Type toTrinoType(IndexMetadata.Field metaDataField)
     {
-        return toTrinoType(metaDataField, metaDataField.isArray());
+        return toTrinoType(metaDataField, metaDataField.asRawJson(), metaDataField.isArray());
     }
 
-    private Type toTrinoType(IndexMetadata.Field metaDataField, boolean isArray)
+    private Type toTrinoType(IndexMetadata.Field metaDataField, boolean asRawJson, boolean isArray)
     {
+        checkArgument(!asRawJson || !isArray, "Single column cannot be declared as a Trino array and also be cast as raw json.");
         IndexMetadata.Type type = metaDataField.getType();
         if (isArray) {
-            Type elementType = toTrinoType(metaDataField, false);
+            Type elementType = toTrinoType(metaDataField, asRawJson, false);
```

Comment:
This does not seem correct.  `isArray` is `true` here, we're passing the wrong argument value.  As far as `asRawJson` is concerned, we can infer it's `false`.
```suggestion
            Type elementType = toTrinoType(metaDataField, false, true);
```